### PR TITLE
supersonic: init at 0.5.2

### LINKED ISSUE #224658: Package request: supersonic
**Project description**

A lightweight cross-platform desktop client for Subsonic music servers (Navidrome, Gonic, Airsonic, etc).

**Metadata**

* homepage URL: https://github.com/dweymouth/supersonic
* source URL: https://github.com/dweymouth/supersonic
* license: GPL3
* platforms: Linux/Mac/Windows

### PR DESCRIPTION
**Pull request by sochotnicky**
## Description of changes

This adds a new package:
A lightweight cross-platform desktop client for Subsonic music servers

Closes https://github.com/NixOS/nixpkgs/issues/224658

Release info: https://github.com/dweymouth/supersonic/releases/tag/v0.5.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sochotnicky** [comment]
Switched to draft, realized the app is running under XWayland so investigating native support for Wayland

**Comment by sochotnicky** [comment]
So unfortunately the underlying library (go-glfw) doesn't seem to support fallback from Wayland to X11. Compiling with Wayland disables X11. So I created 2 versions of this package and made them installable in parallel (with desktop file having a suffix to distinguish them)

**Comment by vs49688** [comment]
Seems we had similar ideas - https://github.com/NixOS/nixpkgs/pull/256466

File: pkgs/applications/audio/supersonic/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    rev = "v${version}";
+    hash = "sha256-4SLAUqLMoUxTSi4I/QeHqudO62Gmhpm1XbCGf+3rPlc=";
+  };
+
+  vendorHash = "sha256-6Yp5OoybFpoBuIKodbwnyX3crLCl8hJ2r4plzo0plsY=";
+
+  nativeBuildInputs = with pkgs; [ pkg-config copyDesktopItems ];
+
+  # go-glfw doesn't support both X11 and Wayland in single build
+  tags = "" + lib.optionalString forceWayland "wayland";
```
Comment (vs49688) [inline pkgs/applications/audio/supersonic/default.nix:20]: `tags` can be a list

```
tags = lib.optionals forceWayland [ "wayland" ]
```

File: pkgs/applications/audio/supersonic/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      categories = [ "Audio" "AudioVideo" ];
+    })
+  ];
+
+  meta = with lib; {
+    description =
+      "A lightweight cross-platform desktop client for Subsonic music servers";
+    homepage = "https://github.com/dweymouth/supersonic";
+    license = licenses.gpl3Plus;
+    platforms = platforms.linux; # Technically MacOS should be supported though
```
Comment (vs49688) [inline pkgs/applications/audio/supersonic/default.nix:59]: It builds on my M1, it just needs `UserNotifications`.

I'm not familiar with the actual Apple packaging process though, so I'm fine to leave it out until someone else wants to to do it.

File: pkgs/applications/audio/supersonic/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    maintainers = with maintainers; [ sochotnicky ];
```
Comment (vs49688) [inline pkgs/applications/audio/supersonic/default.nix:60]: Feel free to add me as a maintainer, use `zane`.

**Comment by vs49688** [comment]
Also, migrate it to https://github.com/NixOS/rfcs/blob/master/rfcs/0140-simple-package-paths.md

Simply rename to `pkgs/applications/audio/supersonic/default.nix` to `pkgs/by-name/su/supersonic/package.nix`, and remove `supersonic` from `all-packages.nix` (keep `supersonic-wayland`)

**Comment by sochotnicky** [comment]
Closing - replaced by https://github.com/NixOS/nixpkgs/pull/257973/

**Pull request closed without merging**